pipeline.go

```diff
@@ -142,10 +151,16 @@ func (c *Pipeline) start() {
 				}
 				req := inFlight.get(a) // match the answer to an in-flight query
 				if req == nil {
+					c.metrics.err.Add("unexpected_a", 1)
 					log.WithField("qname", qName(a)).Warn("unexpected answer received, ignoring")
 					continue
 				}
+				c.metrics.response.Add(rCode(a), 1)
 				req.markDone(a, nil)
+				ql := (int64)(inFlight.maxLen)
```

Comment:
Isn't this a possible race since maxLen isn't locking the mutex in the inFlightQueue? Might be better to add a safe method to the queue to return that value.

Reply:
Addressed in a subsequent change.